### PR TITLE
feat: image preview [WPB-6733]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6733" title="WPB-6733" target="_blank">StoryWPB-6733</a>  Preview image before sending them
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- image preview before sending
- improved handlers for legal hold errors when sending message to be used also in image preview
- added legal hold handler for send ping

### Attachments (Optional)

https://github.com/wireapp/wire-android/assets/13151239/32a8219d-812e-42e6-98dc-99a016ea5e17

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3928](https://10.10.124.134/job/AR-build-pipeline/3928/) **failed**.

**Comment by AndroidBob** [comment]
Build [3929](https://10.10.124.134/job/AR-build-pipeline/3929/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9317552)(9317552)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9317552)(9317552)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [3943](https://10.10.124.134/job/AR-build-pipeline/3943/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/sendmessage/SendMessageViewModel.kt [hunk 1]
```diff
@@ -273,10 +298,14 @@ class SendMessageViewModel @Inject constructor(
...
 
-    private fun ScheduleNewAssetMessageResult.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) = also {
+    private fun ScheduleNewAssetMessageResult.handleLegalHoldFailureAfterSendingMessage(conversationId: ConversationId) = let {
         if (it is ScheduleNewAssetMessageResult.Failure) {
             it.coreFailure.handleLegalHoldFailureAfterSendingMessage(conversationId)
         }
+        when (this) {
+            is ScheduleNewAssetMessageResult.Failure -> Either.Left(coreFailure)
+            is ScheduleNewAssetMessageResult.Success -> Either.Right(Unit)
+        }
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/sendmessage/SendMessageViewModel.kt:308]: doesn't the if above cancels the execution of the Failure inside `when`? 🤔

**Review by alexandreferris — COMMENTED** [review]
Great work! just a tiny question regarding the code execution with both `if` and `when` statements

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/sendmessage/SendMessageViewModel.kt (hunk 1 above)
Comment (alexandreferris) [reply app/src/main/kotlin/com/wire/android/ui/home/conversations/sendmessage/SendMessageViewModel.kt:308]: I thought `it.coreFailure.handleLegalHoldFailureAfterSendingMessage(conversationId)` was returning something, but its not so the flow is fine 😓

**Comment by AndroidBob** [comment]
Build [4053](https://10.10.124.134/job/AR-build-pipeline/4053/) **failed**.

**Comment by AndroidBob** [comment]
Build [4055](https://10.10.124.134/job/AR-build-pipeline/4055/) **failed**.

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4070](https://10.10.124.134/job/AR-build-pipeline/4070/) **failed**.

**Comment by AndroidBob** [comment]
Build [4085](https://10.10.124.134/job/AR-build-pipeline/4085/) **failed**.

**Review by mchenani — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4100](https://10.10.124.134/job/AR-build-pipeline/4100/) **failed**.

**Comment by AndroidBob** [comment]
Build [4107](https://10.10.124.134/job/AR-build-pipeline/4107/) **failed**.

**Pull request merged**